python/numpy.h

```diff
@@ -12,24 +13,89 @@
 #include "pybind11.h"
 
 namespace py = pybind11;
+
 using namespace scipp;
 
-template <class T, class View>
+/// Map C++ types to Python types to perform conversion between scipp containers
+/// and numpy arrays.
+template <class T> struct ElementTypeMap {
+  using CppType = T;
+  using PyType = T;
+  constexpr static bool convert = false;
+
+  static void check_assignable(const py::object &, const units::Unit &) {}
+};
+
+template <> struct ElementTypeMap<scipp::core::time_point> {
+  using CppType = scipp::core::time_point;
+  using PyType = int64_t;
+  constexpr static bool convert = true;
+
+  static void check_assignable(const py::object &obj, units::Unit unit);
+};
+
+template <bool convert, class Source, class Destination>
+void copy_element(const Source &src, Destination &&dst) {
```

Comment:
```suggestion
constexpr void copy_element(const Source &src, Destination &&dst) noexcept {
```
or is it not?

Reply:
Are `ElementArrayView`s constexpr, noexcept copyable? This function does not only operate on fundamental values.